src/components/editor/ImportResourceTemplate.jsx

```diff
@@ -85,20 +85,44 @@ class ImportResourceTemplate extends Component {
   updateStateFromServerResponse = (response, profileCount, updateKey) => {
     // HTTP status 409 == Conflict
     const showModal = response.status === 409 && this.state.createResourceError.length >= profileCount
-
-    const location = response.headers.location || ''
-    const newMessage = `${this.humanReadableStatus(response.status)} ${location}`
-    const newState = {
-      message: [...this.state.message, newMessage],
-      updateKey: updateKey
-    }
+    const newMessage = `${this.humanReadableStatus(response.status)} ${this.humanReadableLocation(response)}`
+    const newState = { updateKey: updateKey }
 
     if (showModal)
       newState.modalShow = true
 
+    // `response.ok` is a Boolean that returns true if the HTTP status is in the
+    // 200-299 range, indicating the HTTP request was successful
+    if (response.ok) {
+      // Wipe out past messages and errors if response was successful
+      newState.createResourceError = []
+      newState.message = [newMessage]
+    } else {
+      // Response not OK, so append message to list of current messages
+      newState.message = [...this.state.message, newMessage]
+    }
+
     this.setState(newState)
   }
 
+  // Returns a URL or an empty string
+  humanReadableLocation = response => {
+    // Use Location header if it exists
+    if (response?.headers?.location)
+      return response.headers.location
+
+    // Use ID in request body if it exists, appending it to the request URL
+    if (response?.req?._data?.id) {
+      // But if the request URL already contains the ID, don't bother appending
+      if (response.req.url.endsWith(response.req._data.id))
+        return response.req.url
+      return `${response.req.url}/${response.req._data.id}`
+    }
+
+    // Welp, we tried anyway
+    return ''
```

Comment:
Any specific reason you're returning an empty string? Does the calling method or function require a string?

Reply:
The calling method uses the returned value when building a string (a label for human consumption in the UI). In general, I try to keep return values consistently typed, though it's not always feasible in practice.

Reply:
@jermnelson Do you think this function could be improved by returning some other value (`null`? `undefined`?) and letting the calling function deal with those values?

Reply:
Not sure but returning an empty string makes sense as you're building a label in the UI; if the calling function needs to react differently to the `null` then I would return null. If you remove the return statement altogether the return value in the calling function would automatically be `undefined`.